### PR TITLE
[MI-1824] display single order

### PR DESCRIPTION
**Pull request by joseconsador**
/cc @zendesk/mintegrations @eddiesholl 

### Description

Display a single order when the Shopify Order ID custom field is populated with a number value.

### References
* JIRA: https://zendesk.atlassian.net/browse/MI-1824

### Risks
* [medium] Differences in API response could cause failure across products.

### CODE REVIEW

File: src/Actions/AppActions.js [hunk 1]
```diff
@@ -63,15 +64,43 @@ export const initLocale = zafClient => (dispatch) => {
   }
 
   return zafClient.get('currentUser.locale').then(
-      localeData => loadTranslationsForLocale(localeData['currentUser.locale']),
+      (localeData) => {
+        let locale = localeData['currentUser.locale'];
+        if (locale === undefined) {
```
Comment (kcasas) [inline src/Actions/AppActions.js]: Is there a reason why you didn't include this condition on `translations.loadTranslations`?

File: src/Middlewares/CustomerResponseMiddleware.js [hunk 2]
```diff
@@ -9,7 +10,8 @@ const customerResponseMiddleware = store => next => (action) => {
     }
 
     const id = action.response.customers[0].id;
-    store.dispatch({ type: FETCH_ORDERS, id });
+
+    store.dispatch(fetchOrders(id));
```
Comment (kcasas) [inline src/Middlewares/CustomerResponseMiddleware.js]: Can be refactored to:
```
if (store.getState().orderId) {
    store.dispatch(fetchOrder(store.getState().orderId));
}
else {
    store.dispatch(fetchOrders(id));
}
```

File: src/Actions/OrdersActions.js [hunk 3]
```diff
@@ -1,8 +1,21 @@
...
+import { FETCH_ORDERS, FETCH_ORDER, FETCH_ORDER_SUCCESS, FETCH_ORDERS_SUCCESS } from './constants';
 
-const fetchOrders = customerId => ({
-  type: FETCH_ORDERS,
-  customerId,
-});
+const fetchOrders = customerId => (dispatch, getState) => {
+  const { orderId } = getState().app;
+
+  if (orderId !== null) {
```
Comment (kcasas) [inline src/Actions/OrdersActions.js]: I think action creators should only be responsible for creating one action. You can create another action creator named `fetchOrder(id)`. And on `CustomerResponseMiddleware` implement something like [this](https://github.com/zendesk/shopify_app/pull/114/files#r139056627)

Comment (joseconsador) [reply src/Actions/OrdersActions.js]: The middleware approach can also be used and would work just the same. But, for me action creators should be able to return whatever type of action it thinks is needed based on the input, that's why they're called creators. Also with this approach you are always sure that an action whether FETCH_ORDER or FETCH_ORDERS will be returned and not have to look at middleware where the action is dispatched but on the side.

File: src/Actions/AppActions.js (hunk 1 above)
Comment (joseconsador) [reply src/Actions/AppActions.js]: This action creator also sets the locale in the app state, so the locale should not be undefined at this point - `translations.loadTranslations` is not aware of that.

**Pull request merged**